metricbeat/module/kafka/broker.go

```diff
@@ -290,6 +295,68 @@ func (b *Broker) FetchPartitionOffsetFromTheLeader(topic string, partitionID int
 	return offset, nil
 }
 
+// FetchPartitionOffsetsForTopics fetches offsets of all partitions from its leaders.
+func (b *Broker) FetchPartitionOffsetsForTopics(topics []*sarama.TopicMetadata, time int64) map[string]map[int32]*PartitionOffsets {
```

Comment:
Nit. It may be good to refactor this method into smaller ones, for example one to get the partitions per broker, another one to make a request of multiple partitions...

Reply:
Refactored, fixed!

Reply:
Great, thanks!